test/TestBuilder.cs

```diff
@@ -343,7 +344,15 @@ private static bool GetNoBuild()
         process.BeginOutputReadLine();
         process.BeginErrorReadLine();
 
-        process.WaitForExit();
+        // Process.WaitForExit() may hang when redirecting output because it actually waits for the
```

Comment:
>// [](http://example.com/codeflow?start=8&length=2)

This is a great finding! Lately our Ubuntu builds started to hang all the time.
I believe it happened after the locks were removed from the result reporting. I was wondering if the cause was the corruption of logWriter caused by accessing from multiple threads.

Reply:
I had removed the locks because I thought maybe they were contributing to the hang. That didn't help, though I didn't think it made anything worse. Anyway it should be good now.